### ISSUE TITLE
[TF2] Game doesn't start on Intel HD 4600

### ISSUE THREAD

**Issue by thedarkfighter413** [ValveSoftware/Source-1-Games#2244]
I usually play on my dedicated graphics card but today I tried to run TF2 on my integrated Intel 4600 and found out it crashes every time before I get to the main menu.

Note: I would add HW/SW info from Steam but it's not possible because my Steam crashes when I open "system information" window but it has nothing to do with this issue

**Steps to reproduce**
1) Get a computer with an Intel HD 4600
2) Start TF2

**Hardware**
CPU: Intel Core i5 4210H
GPU: Intel HD 4600
RAM: 16GB DDR3 1600MHz
Laptop: yes

**Software**
OS: Arch Linux 64bit
DE: Gnome Shell 3.24
Kernel: 4.10.11-1-ARCH #1 SMP PREEMPT Tue Apr 18 08:39:42 CEST 2017 x86_64 GNU/Linux

**GPU drivers**
extra/mesa 17.0.4-2 [installed]
extra/vulkan-intel 17.0.4-2 [installed]
multilib/lib32-mesa 17.0.4-2 [installed]
extra/vulkan-intel 17.0.4-2 [installed]

**What did I try**
1) Launching another Source 1 game (CSGO): worked with no problems
2) Rebooting the computer
3) Verifying integrity
4) Adding "-safe" launch parameter

**Terminal Output**
https://gist.github.com/thedarkfighter413/720306478cfa71796059aa89479883e7#file-gistfile1-txt

I hope this helps. I don't usually play on my integrated GPU so I could just ignore it and switch to my dedicated NVIDIA card but I want to help to make Linux gaming better. Tell me if you need anything else (info, terminal outputs etc.).

EDIT:
integrategratingrated GPU -> integrated GPU

**Comment by kisak-valve** [comment]
Hello @thedarkfighter413, are you running steam with the steam runtime or with Arch's native runtime?

Are you able to run TF2 with your discrete video card to check the graphics settings? If you can, please include what they are set to and test if the game starts if those settings are set to low. I suspect TF2 is running out of virtual address space (4GB inherent to all 32 bit applications) which is used by both the game engine and mesa for graphics (#2212).

If that does not help, then please add any relevant minidumps from `/tmp/dumps/` to an archive and drag and drop the archive into the comment box.

**Comment by thedarkfighter413** [comment]
@kisak-valve Steam runtime. 
I switched to my dedicated NVIDIA card and started the game. Everything was set to very high with 2X MSAA so I set everything to low and disabled AA. And now the game starts correctly on Intel GPU but I can't join any game.

![snimek z 2017-04-28 19-01-56](https://cloud.githubusercontent.com/assets/28152817/25539120/e2fce40e-2c45-11e7-8ab8-5e54735c2e26.png)

Terminal output here https://gist.github.com/thedarkfighter413/416c1b2e97f887edb460a59a4d844fc0#file-gistfile1-txt
